src/nupic/algorithms/Connections.cpp

```diff
@@ -330,7 +332,8 @@ Activity Connections::computeActivity(const vector<Cell>& input,
       // TODO: Possible optimization - define constant variable here?
       synapseData = dataForSynapse(synapse);
 
-      if (synapseData.permanence >= permanenceThreshold)
+      // Add epsilon to permanence before comparing to avoid rounding edge cases
+      if (synapseData.permanence + EPSILON >= permanenceThreshold)
```

Comment:
Yes, that's correct. Subutai and I talked about this, and I can't think of a good way to solve the fundamental problem. This change addresses the issue for the most common choices of `permanenceThreshold`.